homeassistant/components/history/__init__.py

```diff
@@ -120,6 +120,11 @@ async def get(
         entity_ids = None
         if entity_ids_str:
             entity_ids = entity_ids_str.lower().split(",")
+            for entity_id in entity_ids:
+                if not hass.states.get(entity_id) or not valid_entity_id(entity_id):
+                    return self.json_message(
+                        "Invalid filter_entity_id", HTTPStatus.BAD_REQUEST
+                    )
```

Comment:
Now that I think about it, the regex check in the REST API is redundant when we also check the state machine, since I think we can assume that if an entitiy_id is in the state machine it must be valid, so doing a regex check only makes sense for the Web Socket API where you can subscribe to entity_ids which don't exist yet.